### DISCUSSION TITLE
Check for Table Scans

### DISCUSSION THREAD

**Discussion by oseibonsu** [oguimbal/pg-mem#262] [Q&A] [answered]
Is it possible to have pg-mem throw an error with any query that requires a complete table scan? I have a specific use case where want to make sure that all queries are using an index. This would allow us to ensure during testing that no one has created a poorly written SQL statement.

**Comment by oseibonsu** [comment, marked as answer]
This question has already been answered in the docs.

**Reply by oguimbal** [reply]
yes :) but beware: pg-mem use of indices is far less advanced compared to postgres => It almost only works on simple where statements and on join conditions. 

 => PG might find a way to use an index, even when pg-mem tells you that it didnt.
